src/Metadata/Property/Factory/CachedPropertyMetadataFactory.php

```diff
@@ -28,6 +28,7 @@
 
     private $cacheItemPool;
     private $decorated;
+    private $memoryCache = [];
```

Comment:
`$memoryCache` is not a very descriptive name... :disappointed:

Perhaps `$localCache` and `$localCacheKey`. Descriptive is better.